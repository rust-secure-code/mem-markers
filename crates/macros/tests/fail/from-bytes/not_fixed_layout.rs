use mem_markers::*;

#[derive(FromBytes, ByteComplete)]
#[repr(C)]
struct Foo {
    a: u8,
}

fn main() {}
