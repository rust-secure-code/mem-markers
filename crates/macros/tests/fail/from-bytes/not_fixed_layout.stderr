error[E0277]: the trait bound `Foo: mem_markers::fixed_layout::FixedLayout` is not satisfied
 --> $DIR/not_fixed_layout.rs:3:10
  |
3 | #[derive(FromBytes, ByteComplete, Zeroable)]
  |          ^^^^^^^^^ the trait `mem_markers::fixed_layout::FixedLayout` is not implemented for `Foo`
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
